mongo_connector/oplog_manager.py

```diff
@@ -183,12 +196,42 @@ def oplog_ns_set(self):
     def update_oplog_ns_set(self):
         self._oplog_ns_set = []
         if self.namespace_set:
-            self._oplog_ns_set.extend(self.namespace_set)
+            for ns in self.namespace_set:
+                if "*" in ns:
+                    # if namespace_set contains *, change to regexp
+                    reg_ns = re.sub(r"(\.)","\.", ns)
+                    reg_ns = re.sub(r"(\*)", ".*", reg_ns)
+                    LOG.debug("OplogThread: Replace namespace %s to %s"  %(ns, reg_ns))
+                    reg = re.compile(reg_ns)
+                    self._oplog_ns_set.append(reg)
+                else:
+                    self._oplog_ns_set.append(ns)
+                    self._oplog_ns_set.append(ns.split('.', 1)[0] + '.$cmd')
+
             self._oplog_ns_set.extend(self.gridfs_files_set)
-            self._oplog_ns_set.extend(set(
-                ns.split('.', 1)[0] + '.$cmd' for ns in self.namespace_set))
             self._oplog_ns_set.append("admin.$cmd")
 
+    @property
+    def oplog_ex_ns_set(self):
+        try:
+            return self._oplog_ex_ns_set
+        except AttributeError:
```

Comment:
Hi @llvtt ,

We did not call self.oplog_ex_ns_set - ... in the constructor, but in the get_last_oplog_timestamp() and get_oplog_cursor(). In the constructor, we call self.ex_namespace_set = ....

I just followed the same way as the code handles with oplog_ns_set. I feel complicated about the part of code as well.

What is your opinion? Do I need to remove all try except for oplog_ns_set, gridfs_files_set and  oplog_ex_ns_set?